crux-agent/app/core/providers/lmstudio.py

```
@@ -445,14 +633,28 @@ async def _handle_streaming_response(self, payload: Dict[str, Any]) -> str:
                         if "choices" in data and data["choices"]:
                             delta = data["choices"][0].get("delta", {})
                             if "content" in delta and delta["content"]:
-                                content += delta["content"]
+                                content += delta.get("content", "")
+                            # Track finish_reason if present (end of stream chunk)
+                            if "finish_reason" in data["choices"][0]:
+                                finish_reason_stream = data["choices"][0]["finish_reason"]
+                                if finish_reason_stream:
+                                    last_finish_reason = finish_reason_stream
```

Ruff (0.12.2)

Local variable `last_finish_reason` is assigned to but never used

Remove assignment to unused variable `last_finish_reason`

(F841)